### PR TITLE
Add start point for ForEach and ForEachTxn

### PR DESCRIPTION
**Pull request by itsmontoya**

### CODE REVIEW

File: forEacher.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+func newForEacher(tid string, fn ForEachFn, mw *middleware.MWs, cor bool) *forEacher {
+	var fe forEacher
+	fe.tid = tid
+	fe.fn = fn
+	fe.mw = mw
+	fe.cor = cor
+
+	if tid == "" {
+		fe.state = 1
```
Comment (shahzilabid) [inline forEacher.go]: Hmm could we add some comments here? Sort of confusing as to what tid exactly is and what each state int represents

File: forEacher.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+		if fe.state != 1 {
+			return
+		}
+
+		var b []byte
+		if fe.mw != nil {
+			var r io.Reader
+			if r, err = fe.mw.Reader(buf); err != nil {
+				return true
```
Comment (shahzilabid) [inline forEacher.go]: Should maybe have journaler print out these errs

File: forEacher.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+	"github.com/itsmontoya/middleware"
+	"github.com/missionMeteora/journaler"
+	"github.com/missionMeteora/uuid"
+)
+
+// forEachState represents the match state of a ForEach/ForEachTx request
+type forEachState uint8
+
+const (
+	// statePreMatch represents an interation which has not yet matched the target transaction
```
Comment (shahzilabid) [inline forEacher.go]: Grammar police.. iteration*. Same with comment blow

File: forEacher.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+	mw  *middleware.MWs
+	cor bool
+	// Match state
+	state forEachState
+}
+
+func (fe *forEacher) processLine(buf *bytes.Buffer) (err error) {
+	var (
+		lineType byte
+		key      []byte
```
Comment (shahzilabid) [inline forEacher.go]: Nit picky/OCD but can be one line.. var key, value []byte

File: forEacher.go [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+	// Note: We will be replacing this with the other new function at the end of this case
+	fe.fn(fe.ti)
+	fe.ti = nil
+}
+
+func (fe *txnForEacher) processLine(buf *bytes.Buffer) (err error) {
+	var (
+		lineType byte
+
+		tid   []byte
```
Comment (shahzilabid) [inline forEacher.go]: Can we make all these []bytes a one line init like I mentioned earlier? If you're against it that's cool too

File: forEacher.go (hunk 3 above)
Comment (itsmontoya) [reply forEacher.go]: HAH, nice catch

File: forEacher.go (hunk 5 above)
Comment (itsmontoya) [reply forEacher.go]: I usually don't like it because it makes the left column wide, let me think about it

**Review by shahzilabid — APPROVED** [review]
Great work as always

**Comment by itsmontoya** [comment]
Thanks for the review!

**Pull request merged**